### PR TITLE
split `requirements-dev.txt` into first and third party dev dependencies

### PR DESCRIPTION
**Pull request by zacharyburnett**
This PR separates `requirements-dev.txt` into 
- `requirements-dev-thirdparty.txt` - third party dependencies (`numpy`, `scipy`, etc)
- `requirements-dev-st.txt` - STScI projects directly related to JWST (`stdatamodels`, `stcal`, `stpipe`, etc.)

**Checklist for maintainers**
- [ ] ~added entry in `CHANGES.rst` within the relevant release section~
- [ ] ~updated or added relevant tests~
- [ ] ~updated relevant documentation~
- [ ] ~added relevant milestone~
- [x] added relevant label(s)
- [ ] ran regression tests, post a link to the Jenkins job below.
      [How to run regression tests on a PR](https://github.com/spacetelescope/jwst/wiki/Running-Regression-Tests-Against-PR-Branches)
- [ ] ~Make sure the JIRA ticket is [resolved properly](https://github.com/spacetelescope/jwst/wiki/How-to-resolve-JIRA-issues)~

### CODE REVIEW

**Comment by nden** [comment]
@zacharyburnett Since this change the Jenkins file could you run a regression test?

File: JenkinsfileRT_dev [hunk 1]
```diff
@@ -74,12 +74,11 @@ bc0.conda_ver = '22.11.1'
 bc0.conda_packages = [
     "python=${python_version}",
 ]
-bc0.pip_reqs_files = ['requirements-dev.txt']
+bc0.pip_reqs_files = ['requirements-dev-st.txt', 'requirements-dev-st.txt']
```
Comment (mairanteodoro) [inline JenkinsfileRT_dev:77]: Is `bc0.pip_reqs_files` supposed to contain the same item twice?

**Review by mairanteodoro — COMMENTED** [review]
Looks good to me.

File: JenkinsfileRT_dev (hunk 1 above)
Comment (zacharyburnett) [inline JenkinsfileRT_dev:77]:
```suggestion
bc0.pip_reqs_files = ['requirements-dev-st.txt', 'requirements-dev-thirdparty.txt']
```

Comment (zacharyburnett) [reply JenkinsfileRT_dev]: Good catch!

**Pull request merged**